### PR TITLE
fix: added logic to capture full url for ga4 page_location param

### PR DESCRIPTION
**Pull request by mihir-4116**
## PR Description

Please include a summary of the change along with the relevant motivation and context.

## Notion ticket

Ticket link

## Screenshots

Please add screenshots for any new features or UI bug fixes for the following browsers -

- Chrome
  >
- Firefox
  >
- Safari
  >

## Security

- [ ] The code changed/added as part of this pull request won't create any security issues with how the software is being used.

### CODE REVIEW

File: src/integrations/GA4/utils.js [hunk 1]
```diff
@@ -173,6 +173,8 @@ function getDestinationEventProperties(
     props,
     contextOp
   );
+  // overriding js sdk page param to capture full url
+  propsWithCustomFields.page_location = window.location.href;
```
Comment (utsabc) [inline src/integrations/GA4/utils.js:177]: Can we not update in the mapping part?

Comment (mihir-4116) [reply src/integrations/GA4/utils.js:177]: @utsabc this is basically coming from analytics page call so we need to override it here

Comment (utsabc) [reply src/integrations/GA4/utils.js:177]: But where are we mapping in `propsWithCustomFields` prior to this?

Comment (mihir-4116) [reply src/integrations/GA4/utils.js]: updated as we discussed

**Review by utsabc — APPROVED** [review]

**Comment by utsabc** [comment]
Add some unit test case for this need not be exhaustive

**Review by koladilip — APPROVED** [review]

**Pull request merged**